.github/workflows/cli-release.yaml

```diff
@@ -0,0 +1,68 @@
+# This defines a workflow to build and release a new version of the aptos CLI.
+# In order to trigger it go to the Actions Tab of the Repo, click "Release CLI" and then "Run Workflow".
+
+name: "Release CLI"
+
+on:
+  workflow_dispatch:
+    inputs:
+      release_tag:
+        type: string
+        required: true
+        description: "The release tag to create. E.g. v0.2.3:"
+      source_git_ref_override:
+        type: string
+        required: false
+        description: "GIT_SHA1_OVERRIDE: Use this to override the Git SHA1, branch or tag to build the binaries from. Defaults to the workflow Git REV, but can be different than that:"
+
+jobs:
+  build-linux-binary:
+    name: "Build Linux binary"
+    runs-on: "ubuntu-latest"
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.inputs.source_git_ref_override }}
+      - name: Build CLI
+        run: scripts/cli/build_cli_release.sh
+      - name: Upload Binary
+        uses: actions/upload-artifact@v3
+        with:
+          name: cli-builds
+          path: aptos-cli-*.zip
+
+  build-os-x-binary:
+    name: "Build OS X binary"
+    runs-on: macos-latest
```

Comment:
Does this compile for macos arm?  or macos intel?

Reply:
It compiles only macos intel at the moment. I can run the intel binary on my arm Mac successfully, so not a huge issue. I can make some enhancement later to also build the arm binaries seperately, but I think since we don't do any perf critical with the CLI, it's not a priority to build separate ARM binaries at this moment.